ui/src/app/pages/data/data-set/export-dataset-modal.tsx

```diff
@@ -138,53 +145,119 @@ export const ExportDatasetModal = ({
         'The request cannot be completed. Please try again or contact Support in the left hand navigation'
       );
     }
-  }
+  };
+
+  const isNotebooksLoading = existingNotebooks === undefined;
+  const isCodePreviewLoading = showCodePreview && !codePreview;
 
-  function loadHtmlStringIntoIFrame(html) {
+  // Any state that necessiatates disabling all modal controls.
+  const modalLoading =
+    isNotebooksLoading ||
+    loadingNotebook ||
+    loadingClipboard ||
+    loadingCode ||
+    isExporting;
+
+  const resetCode = () => {
+    setCodePreview(null);
+    setCodeText(null);
+  };
+
+  const onChangeGenomicsAnalysisTool = (
+    genomicsTool: DataSetExportRequestGenomicsAnalysisToolEnum
+  ) => {
+    resetCode();
+    setGenomicsAnalysisTool(genomicsTool);
+  };
+  const genomicsToolRadioButton = (
+    displayName: string,
+    genomicsTool: DataSetExportRequestGenomicsAnalysisToolEnum
+  ) => {
+    return (
+      <label
+        key={'genomics-tool-' + genomicsTool}
+        style={styles.radioButtonLabel}
+      >
+        <RadioButton
+          name='genomics-tool'
```

Comment:
this needs to include the tool name as well, like the other fields

Reply:
I believe that this is supposed to work differently for radio buttons in order to define a radio group:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio#defining_a_radio_group

Reply:
Thanks!  I was following `disk-selector` which now I see may be incorrect!  I'll take a look at that.  Actually, that should probably not be a radio button at all...

Reply:
WIP on that is https://github.com/all-of-us/workbench/pull/8251 - TODO in Jan